ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/snapshot/services/SnapshotServiceImpl.java

```diff
@@ -25,28 +24,27 @@ public class SnapshotServiceImpl implements SnapshotService {
 	private final LdesFragmentRepository ldesFragmentRepository;
 	private final SnapshotRepository snapshotRepository;
 	private final SnapshotRelationLinker snapshotRelationLinker;
-	private final ViewService viewService;
+	private final EventStreamService eventStreamService;
 
 	public SnapshotServiceImpl(SnapShotCreator snapShotCreator, LdesFragmentRepository ldesFragmentRepository,
 			SnapshotRepository snapshotRepository, SnapshotRelationLinker snapshotRelationLinker,
-			ViewService viewService) {
+			EventStreamService eventStreamService) {
 		this.snapShotCreator = snapShotCreator;
 		this.ldesFragmentRepository = ldesFragmentRepository;
 		this.snapshotRepository = snapshotRepository;
 		this.snapshotRelationLinker = snapshotRelationLinker;
-		this.viewService = viewService;
+		this.eventStreamService = eventStreamService;
 	}
 
 	@Override
 	public void createSnapshot(String collectionName) {
 		Optional<Snapshot> lastSnapshot = retrieveLastSnapshot();
 
-		ViewName viewName = viewService.getViewsByCollectionName(collectionName).stream()
-				.map(ViewSpecification::getName)
-				.filter(name -> name.equals(new ViewName(collectionName, ViewConfig.DEFAULT_VIEW_NAME)))
-				.findFirst()
-				.orElseThrow(() -> new SnapshotCreationException(
-						"No default pagination view configured for collection " + collectionName));
+		if (!eventStreamService.retrieveEventStream(collectionName).isDefaultViewEnabled()) {
```

Comment:
This is not really correct. Snapshotting would still be possible if default view is disabled but you manually create a view with DEFAULT_VIEW_NAME.